include/mbedtls/mbedtls_config.h

```diff
@@ -3634,8 +3634,23 @@
 
 /* Platform options */
 //#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h> /**< Header to include if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is defined. Don't define if no header is needed. */
-//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined */
-//#define MBEDTLS_PLATFORM_STD_FREE            free /**< Default free to use, can be undefined */
+/** \def MBEDTLS_PLATFORM_STD_CALLOC
+ *
+ * Default allocator to use, can be undefined.
```

Comment:
But what happens if it is undefined? Is `calloc()` used by default? Is no memory allocation done?

Reply:
i.e. it would be great if this said something like "Default allocator to use, if left undefined, system `calloc()` will be used" (or whatever is actually correct!)

Reply:
Good question. It's complicated (more than it should be, in my opinion — I'd like to simplify it but it would be a backward incompatible change). Incidentally, documenting that complexity would be good: we currently don't detail the interactions between several configuration macros. The details are in `include/mbedtls/platform.h` and `library/platform.c`.

* If one of `MBEDTLS_PLATFORM_C` or `MBEDTLS_PLATFORM_MEMORY` is not defined, then the `STD` macros are ignored. Mbed TLS calls the standard library's `calloc` and `free`.
* If all of the macros `MBEDTLS_PLATFORM_C`, `MBEDTLS_PLATFORM_MEMORY` , `MBEDTLS_PLATFORM_CALLOC_MACRO` and `MBEDTLS_PLATFORM_FREE_MACRO` are defined, Mbed TLS calls `MBEDTLS_PLATFORM_CALLOC_MACRO` and `MBEDTLS_PLATFORM_FREE_MACRO`. The `STD` macros are ignored.
* If `MBEDTLS_PLATFORM_C` and `MBEDTLS_PLATFORM_MEMORY` are defined but at least one of the `{CALLOC,FREE}_MACRO` macros is undefined, then the `STD` macros are used. Furthermore:
    * If `MBEDTLS_PLATFORM_NO_STD_FUNCTIONS` is defined, then `MBEDTLS_PLATFORM_STD_CALLOC` and `MBEDTLS_PLATFORM_STD_FREE` default to `calloc` and `free` respectively. 
    * Otherwise they default to functions that do nothing and (for the calloc one) return `NULL`.

(I hope I got this right. Please double-check!)

Reply:
I think that it would be better to present it like a logic tree. See my proposal in the newest commits.